src/Parser/Typecast.php

```diff
@@ -41,7 +41,7 @@ public function cast(array $values): array
     {
         try {
             foreach ($this->rules as $key => $rule) {
-                if (!array_key_exists($key, $values) || $values[$key] === null) {
+                if (!isset($values[$key])) {
```

Comment:
`!isset()` yields exaclty the result as `!array_key_exists($key, $values) || $values[$key] === null`, but faster (pinging @wolfy-j )

Reply:
Confirmed. I can't remember how we ended with a condition like that, most likely the second part was added on a later stage. Thank you for catching it, it should affect all the persist login in a good way.